### DISCUSSION TITLE
Overriding Tailwind CSS Classes in a Reusable VueJS Component

### DISCUSSION THREAD

**Discussion by thecodeclimber** [tailwindlabs/tailwindcss#1951] [Help]
I have created a VueJS button component using TailwindCSS.  My goal is to provide that button component with some base styles (using tailwindcss classes), with the option to override them when need be (again, using tailwind css classes).  

For example, here is a simplified version of the `Button` component:

```
// Button.vue

<template>
  <button class="bg-green-500 text-white py-2 px-4 rounded">
    Click Me
  </button>
</template>
```

And here is an example of my using that component in another file:

```
// index.vue

<Button></Button>
<Button class="bg-red-600"></Button>
<Button class="bg-blue-600"></Button>
```

The problem is that this only half-works.  That is to say, `bg-blue-600` does override the `bg-green-500` that I set up as a default in `Button.vue`.  But `bg-red-600` does **not** override the background color (presumbably, because `bg-red-600` comes earlier in the css source code.

As such, I am wondering how can I properly set this up?  That is to say, how can I give the `Button` component some base styles (again, using tailwind css classes) while also providing the option to override those styles using tailwind css classes.

Thanks.

**Comment by estevanmaito** [comment]
I'm doing this for React and the approach I'm following is to provide an external theme config, and if people want to change any of the base styles, they can do it there. The components then get their styles from that file. like

```js
button-base-style: 'py-2 px-6 rounded-lg',
button-primary-color: 'bg-purple-600 text-white',
```

In my case, I only provide one colored button, so there is not much change. If you think your component will have many styles, I would provide just a base style button, and then people add their styles. If the same custom component is then used frequently, it can be encapsulated like `RedButton`, `DangerButton`, etc

**Comment by mariusmarais** [comment]
@thecodeclimber I'm dealing with this same issue in React and published an experimental package yesterday to handle these overrides if you want to check it out: https://github.com/mariusmarais/tailwind-cascade

**Comment by saifobeidat** [comment]
Proper fix for this issue:
https://dev.to/saifobeidat/override-tailwind-classes-of-a-reusable-vue-component-without-using-props-660
